### PR TITLE
extend mouse api with move and leave

### PR DESCRIPTION
**Pull request by regeldso**

### CODE REVIEW

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			const { status, target, code } = state;
+			const span = document.createElement('span');
+			let targetString = 'null';
+			if (target) {
+				const { columnIndex, rowIndex } = target;
+				targetString = `{ column: ${columnIndex}, row: ${rowIndex} }`;
+
+			}
+
+			span.innerHTML = `status: ${status},<br>
```
Comment (klumba12) [inline src/examples/mouse-api-basic/example-mouse-api-basic.component.ts:48]: use angular way please don't use innerHTML

Comment (regeldso) [reply src/examples/mouse-api-basic/example-mouse-api-basic.component.ts]: updated

File: projects/qgrid-core/mouse/mouse.state.d.ts [hunk 2]
```diff
@@ -1,7 +1,13 @@
 import { Td } from '../dom/td';
 
 export declare type MouseStateCode = 'left' | 'right' | 'middle' | null;
-export declare type MouseStateStatus = 'release' | 'down' | 'up';
+export declare type MouseStateStatus =
+	| 'release'
```
Comment (klumba12) [inline projects/qgrid-core/mouse/mouse.state.d.ts:5]: is it a valid syntax for the first item?

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<q-grid [caption]="title"
+		[rows]="rows | async">
+	<q-grid-columns generation="deep">
+	</q-grid-columns>
+	<q-grid-toolbar>
+		<ng-template for="right">
+			<h2 class="mat-card-title">Mouse event log</h2>
+			<div id="mouse-event-log"
+				 #mouseEventLog>
+				<span *ngFor="let entry of logEntries">
```
Comment (klumba12) [inline src/examples/mouse-api-basic/example-mouse-api-basic.component.html:10]: use ul\li it's common for the lists

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		dataService: DataService,
+		private cdr: ChangeDetectorRef
+	) {
+		this.rows = dataService.getPeople();
+	}
+
+	static id = EXAMPLE_TAGS[0];
+
+	@ViewChild(GridComponent) grid: GridComponent;
+	@ViewChild('mouseEventLog') mouseEventLog: ElementRef;
```
Comment (klumba12) [inline src/examples/mouse-api-basic/example-mouse-api-basic.component.ts:34]: remove this

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+})
+export class ExampleMouseApiBasicComponent implements AfterViewInit {
+	constructor(
+		dataService: DataService,
+		private cdr: ChangeDetectorRef
+	) {
+		this.rows = dataService.getPeople();
+	}
+
+	static id = EXAMPLE_TAGS[0];
```
Comment (klumba12) [inline src/examples/mouse-api-basic/example-mouse-api-basic.component.ts:31]: is it should be on the top of file?

File: src/examples/mouse-api-basic/example-mouse-api-basic.log-entry.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export declare class LogEntry {
```
Comment (klumba12) [inline src/examples/mouse-api-basic/example-mouse-api-basic.log-entry.ts:1]: please define it in the same component file (it's required when we will export this example to the stackblitz)

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.ts (hunk 5 above)
Comment (regeldso) [reply src/examples/mouse-api-basic/example-mouse-api-basic.component.ts:24]: yep

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.html (hunk 3 above)
Comment (regeldso) [reply src/examples/mouse-api-basic/example-mouse-api-basic.component.html]: changed

File: projects/qgrid-core/mouse/mouse.state.d.ts (hunk 2 above)
Comment (regeldso) [reply projects/qgrid-core/mouse/mouse.state.d.ts:5]: this is an auto formating and yes seems its totally fine

File: src/examples/mouse-api-basic/example-mouse-api-basic.component.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		model.mouseChanged.on(({ state }) => {
+			const { status, target, code } = state;
+			let targetString = 'null';
+			if (target) {
+				const { columnIndex, rowIndex } = target;
+				targetString = `{ column: ${columnIndex}, row: ${rowIndex} }`;
+			}
+
+			this.logEntries.unshift({
+				status: status ?? 'null',
```
Comment (klumba12) [inline src/examples/mouse-api-basic/example-mouse-api-basic.component.ts:50]: use real null values and convert to 'null' in the html

**Review by klumba12 — APPROVED** [review]

**Pull request merged**